GVRf/Framework/framework/src/main/java/org/gearvrf/GVRBehavior.java

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.lang.reflect.Method;
+
+/**
+ * Base class for adding user-defined behaviors to a scene object.
+ * You can override callbacks for initialization and per-frame updates.
+ * 
+ * This class listens for draw frame events when it is attached to a scene object.
+ * You can override these callbacks to implement custom components.
+ * - onDrawFrame(float frametime) called once every frame before rendering.
+ * - onAttach(GVRSceneObject) called when this behavior is attached to a scene object.
+ * - onDetach(GVRSceneObject) called when this behavior is detached from a scene object.
+ * 
+ * @see GVRComponent
+ * @see GVRSceneObject.attachComponent
+ * @see GVRSceneObject.getComponent
+ * @see GVRSceneObject.attachComponent
+ * @see GVRSceneObject.detachComponent
+ */
+public class GVRBehavior extends GVRComponent implements GVRDrawFrameListener
+{
+    protected boolean mIsListening;
+    protected long mType;    
+    private boolean mHasFrameCallback;
+    static private long TYPE_BEHAVIOR = (System.currentTimeMillis() & 0xfffffff);
+    
+    /**
+     * Constructor for a behavior.
+     *
+     * @param gvrContext    The current GVRF context
+     * @param nativePointer Pointer to the native object, returned by the native constructor
+     */
+    protected GVRBehavior(GVRContext gvrContext)
+    {
+        this(gvrContext, 0);
+        mType = TYPE_BEHAVIOR;
+    }
+    
+    /**
+     * Constructor for a behavior.
+     *
+     * @param gvrContext    The current GVRF context
+     * @param nativePointer Pointer to the native object, returned by the native constructor
+     */
+    protected GVRBehavior(GVRContext gvrContext, long nativeConstructor)
+    {
+        super(gvrContext, nativeConstructor);
+        mIsListening = false;
+        mHasFrameCallback = isImplemented("onDrawFrame", float.class);
+    }    
+
+    static public long getComponentType()
+    {
+        return TYPE_BEHAVIOR;
+    }
+    
+    public void enable()
```

Comment:
start&stopListening probably needs some synchronization to ensure thread-safety.